app/javascript/components/DonationBands/DonationBands.js

```diff
@@ -28,69 +28,56 @@ export class DonationBands extends Component {
   props: Props;
 
   state: {
-    customAmount: string,
+    customAmount?: number,
   };
 
   constructor(props: Props) {
     super(props);
 
-    let customAmount = '';
-    if (props.customAmount) {
-      customAmount = props.customAmount.toString();
-    }
-
     this.state = {
-      customAmount: customAmount,
+      customAmount: props.customAmount,
     };
   }
 
   onButtonClicked(amount: number = 0) {
-    this.setState({ customAmount: '' });
+    this.setState({ customAmount: undefined });
     this.props.selectAmount(amount);
     this.props.proceed();
   }
 
   onInputUpdated(value: string) {
-    let amount = null;
-    // \u00a3 is £, \u20ac is €
-    const match = value.match(/^[$\u20ac\u00a3]*(\d{0,10})/);
-
-    if (match && match[1].length) {
-      amount = parseFloat(match[1]);
-      this.setState({ customAmount: amount.toString() });
-    } else if (value === '' || (match && match[1] === '')) {
-      amount = null;
-      this.setState({ customAmount: '' });
-    } else {
-      amount = parseFloat(this.state.customAmount);
-    }
-
+    // Remove non-digit characters before parsing
+    const number = value.replace(/\D/g, '');
+    const amount = number ? parseFloat(number) : undefined;
+    this.setState({ customAmount: amount });
     if (this.props.selectAmount) {
       this.props.selectAmount(amount);
     }
   }
 
   customFieldDisplay() {
-    const amountString = this.state.customAmount || '';
-    if (amountString === '') return '';
-    let currencySymbol = this.props.currency === 'GBP' ? '£' : '$';
-    if (this.props.currency === 'EUR') currencySymbol = '€';
-    return `${currencySymbol}${amountString}`;
+    if (!this.state.customAmount) return '';
+    return this.props.intl.formatNumber(this.state.customAmount, {
+      currency: this.props.currency,
+      style: 'currency',
+      minimumFractionDigits: 0,
+      maximumFractionDigits: 0,
+    });
```

Comment:
We had some custom logic here to display $ for every currency except £ and $. I've updated this to use the default intl formatter since it only affects the custom input box, and it's cleaner.